### PR TITLE
feat: improve test coverage for MCP server layer

### LINKED ISSUE #3: Improve test coverage and instrumentation for MCP server/tool layers
## Summary
Increase test coverage across the project beyond `src/pylon-client.ts`, and ensure instrumentation includes server/MCP tool layers.

## Current state
- Coverage (from coverage-final.json):
  - Statements: 41.1% (30/73)
  - Functions: 35.3% (12/34)
  - Branches: 85.7% (6/7)
  - Lines: 48.7% (56/115)
- Instrumentation appears limited to `src/pylon-client.ts`; server/MCP layers are not covered.

## Proposed work
- Enable full-source instrumentation in Vitest (adjust include patterns/config so server/tool code is covered).
- Add unit/integration tests for MCP server/tool layer (dist/src entry, tool wrappers) including negative paths (missing params, error handling).
- Add additional client tests for error cases and edge conditions (e.g., retries, bad payloads) if applicable.
- Re-run coverage to confirm meaningful improvement.

## Acceptance criteria
- Coverage meaningfully increased (target TBD; at minimum server/tool files included in report).
- Tests verify both success and failure paths for MCP tool handlers.
- Coverage report generated and attached/linked in PR description.

### PR DESCRIPTION
**Pull request by mgmonteleone**
## Summary

This PR addresses issue #3 by improving test coverage for the MCP server layer.

### Configuration
- **vitest.config.ts**: Added `include: ['src/**/*.ts']` to ensure all source files are instrumented for coverage

### New Files
- **src/server-helpers.ts**: Extracted testable helper functions from `index.ts`:
  - `parseCacheTtl()` - Parse and validate PYLON_CACHE_TTL
  - `isMessageConfirmationRequired()` - Check if message confirmation is required
  - `createPylonClient()` - Create PylonClient instance
  - `ensurePylonClient()` - Ensure client is initialized
  - `jsonResponse()` - Create MCP tool JSON response
  - `processElicitationResult()` - Process elicitation result for message confirmation
  - `buildElicitationMessage()` - Build elicitation message
  - `buildElicitationSchema()` - Build elicitation schema

- **tests/server-helpers.test.ts**: 36 comprehensive tests for all helper functions (100% coverage)

- **tests/mcp-server.test.ts**: 21 tests for MCP server patterns and logic

### Modified Files
- **src/index.ts**: Refactored to use helper functions from `server-helpers.ts`
- **tests/pylon-client.core.test.ts**: Added 10 error handling tests (401, 500, timeout, POST/PATCH/DELETE errors)
- **tests/pylon-mcp.functional.tools.test.ts**: Added 9 tests for additional tools

## Coverage Results

| Metric | Before | After | Change |
|--------|--------|-------|--------|
| Statements | 41.1% | 60.07% | **+18.97%** |
| Branches | 85.7% | 82.41% | -3.29%* |
| Functions | 35.3% | 59% | **+23.7%** |
| Lines | 48.7% | 59.62% | **+10.92%** |

*Branch coverage decreased slightly because more files are now included in the report, but the absolute number of covered branches increased.

### File-Level Coverage

| File | Statements | Branches | Functions | Lines |
|------|------------|----------|-----------|-------|
| server-helpers.ts | 100% | 100% | 100% | 100% |
| pylon-client.ts | 93.15% | 87.3% | 96.22% | 93% |
| index.ts | 0% | 0% | 0% | 0% |

**Note**: `index.ts` shows 0% because it has side effects at module load time (starts the server). The testable logic has been extracted to `server-helpers.ts` which has 100% coverage.

## Test Summary

- **Total Tests**: 177 (all passing ✅)
- **New Test Files**: 2

## Acceptance Criteria

- ✅ Coverage report includes `src/index.ts` (and new `src/server-helpers.ts`)
- ✅ Tests verify both success and failure paths for MCP tool handlers
- ✅ Coverage meaningfully increased from ~41% to ~60% statements

Closes #3

---
Pull Request opened by [Augment Code](https://www.augmentcode.com/) with guidance from the PR author

### CODE REVIEW

File: src/server-helpers.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  originalContent: string
+): { confirmed: boolean; content?: string; reason?: string } {
+  if (result.action === 'accept' && result.content) {
+    const confirmSend = result.content.confirm_send as boolean;
+    const modifiedContent = result.content.modified_content as string | undefined;
+
+    if (confirmSend) {
+      return {
+        confirmed: true,
+        content: modifiedContent && modifiedContent.trim() ? modifiedContent.trim() : originalContent,
```
[the remark this answers is not in the document]
Comment (mgmonteleone) [reply src/server-helpers.ts]: Fixed in commit 0cdb4c2.

Added a `typeof` guard to ensure `modified_content` is actually a string before using it:

```typescript
const rawModifiedContent = result.content.modified_content;
const modifiedContent =
  typeof rawModifiedContent === 'string' ? rawModifiedContent : undefined;
```

This prevents runtime errors if the value is unexpectedly a non-string type (e.g., number, null, object) since `content` is typed as `Record<string, unknown>`.

Added tests for:
- Non-string `modified_content` (e.g., number `12345`)
- Null `modified_content`

**Comment by mgmonteleone** [comment]
augment review

File: src/server-helpers.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+ * @param result - The elicitation result from MCP
+ * @param originalContent - The original message content
+ * @returns Confirmation result with confirmed status and content
+ */
+export function processElicitationResult(
+  result: { action: string; content?: Record<string, unknown> },
+  originalContent: string
+): { confirmed: boolean; content?: string; reason?: string } {
+  if (result.action === 'accept' && result.content) {
+    const confirmSend = result.content.confirm_send as boolean;
```
[the remark this answers is not in the document]
Comment (mgmonteleone) [reply src/server-helpers.ts]: Fixed in commit 861529b.

Added strict boolean check for `confirm_send` to require exactly `true` (not just truthy):

```typescript
const rawConfirmSend = result.content.confirm_send;
const confirmSend = rawConfirmSend === true;
```

This prevents truthy non-boolean values like `'false'` string or `1` from allowing message sends.

Added tests for:
- String `'true'` rejection
- String `'false'` rejection  
- Number `1` rejection
